plugins/CoreHome/javascripts/dataTable_rowactions.js

```diff
@@ -373,7 +373,7 @@ DataTable_RowActions_RowEvolution.prototype.showRowEvolution = function (apiMeth
             // remember label for multi row evolution
             box.find('.rowevolution-startmulti').click(function () {
                 Piwik_Popover.onClose(false); // unbind listener that resets multiEvolutionRows
-                Piwik_Popover.close();
+                broadcast.propagateNewPopoverParameter(false);
```

Comment:
Would it be possible that `Piwik_Popover.close()` calls this method or not easily doable? I feel like it would be better understandable to have only `Piwik_Popover.close()` which is quite expressive and something you would expect to call in order to close the popup but as only we are supposed to use it, it might be fine if not doable.

Reply:
Or maybe we could have another method like `Piwik_Popover.closeAll()` or something? I'm not quite sure what the difference is between `Popover.close()` and `broadcast.propagateNewPopoverParameter(false);` and when to call any of them.

Reply:
Not really, at least not w/o a fair amount of work. The problem is that it is used by, eg, `popover-handler.js` to close a popover and that is in the middle of a popover change when we just want to remove the popover element, but not the popover parameter.

I think a lot of this code needs to be reworked later.

Reply:
sounds good to rework it later. Just out of curiosity when needs `broadcast.propagateNewPopoverParameter(false);` called and when `Piwik_Popover.close()` and what do they do differently? be good to explain this somewhere or have a more "speakable" method name for it.

Reply:
Originally `broadcast.propagateNewPopoverParameter(false);` unset the `popover=` query parameter & `Piwik_Popover.close()` would close the popup and then call `broadcast.propagateNewPopoverParameter(false);`.

Now, the responsibilities are:
`broadcast.propagateNewPopoverParameter(false);`: removes the popover query parameter (& triggers the $locationChangeSuccess event through a digest)
`Piwik_Popover.close()`: removes modal (ie, removes the DOM element & cleans up)

So one affects the element and another affects the URL.